docs/REST/authentication.md

```diff
@@ -289,3 +289,68 @@ static class HttpContentExtensions
     }
 }
 ```
+### Java
+```java
+public class ConfigHttpClient {
+
+    public CloseableHttpResponse execute(HttpUriRequest request, String credential, String secret)
+            throws ClientProtocolException, IOException, UnsupportedOperationException, URISyntaxException {
+        Map<String, String> authHeaders = sign(request, credential, secret);
+        authHeaders.forEach(request::setHeader);
+
+        return httpClient.execute(request);
+    }
+
+    private static Map<String, String> sign(HttpUriRequest request, String credential,
+            String secret) throws URISyntaxException, UnsupportedOperationException, IOException {
+        String requestTime = GMT_DATE_FORMAT.format(new Date());
+
+        String contentHash = buildContentHash(request);
+
+        // SignedHeaders
+        String signedHeaders = "x-ms-date;host;x-ms-content-sha256";
+
+        // Signature
+        String methodName = request.getRequestLine().getMethod().toUpperCase();
+        URIBuilder uri = new URIBuilder(request.getRequestLine().getUri());
+        String scheme = uri.getScheme() + "://";
+        String requestPath = uri.toString().substring(scheme.length()).substring(uri.getHost().length());
+        String host = new URIBuilder(request.getRequestLine().getUri()).getHost();
+        String toSign = String.format("%s\n%s\n%s;%s;%s", methodName, requestPath, requestTime, host, contentHash);
+
+        byte[] decodedKey = Base64.getDecoder().decode(secret);
+        String signature = Base64.getEncoder().encodeToString(new HmacUtils(HMAC_SHA_256, decodedKey).hmac(toSign));
+
+        // Compose headers
+        Map<String, String> headers = new HashMap<>();
+        headers.put("x-ms-date", requestTime);
+        headers.put("x-ms-content-sha256", contentHash);
+
+        String authorization = String.format("HMAC-SHA256 Credential=%s, SignedHeaders=%s, Signature=%s",
+                credential, signedHeaders, signature);
+        headers.put("Authorization", authorization);
+
+        headers.put(HttpHeaders.USER_AGENT, USER_AGENT);
```

Comment:
Why do we need this line for signing? A user can always add a user-agent header if they want.